ui/apps/platform/src/hooks/useFetchClusterNamespacesForPermissions.ts

```diff
@@ -0,0 +1,66 @@
+import { useState, useEffect } from 'react';
+import { ScopeObject, getNamespacesForClusterAndPermissions } from 'services/RolesService';
+import { getAxiosErrorMessage } from 'utils/responseErrorUtils';
+
+export type Namespace = {
+    id: string;
+    name: string;
+};
+
+type NamespaceResponse = {
+    loading: boolean;
+    error: string;
+    namespaces: Namespace[];
+};
+
+const emptyResponse: NamespaceResponse = {
+    loading: false,
+    error: '',
+    namespaces: [] as Namespace[],
+};
+
+export function useFetchClusterNamespacesForPermissions(
```

Comment:
I couldn't find usages of this function

Reply:
It is now used in the namespace filter (and in the Network Graph 2.0 page).